src/main/scala/scalismo/faces/parameters/Illumination.scala

```diff
@@ -52,6 +52,11 @@ object DirectionalLight {
 
 /** Spherical Harmonics illumination parameters, environment map */
 case class SphericalHarmonicsLight(coefficients: IndexedSeq[Vector[_3D]]) extends Illumination {
+  require(coefficients.isEmpty || SphericalHarmonics.totalCoefficients(bands) == coefficients.length, "invalid length of IndexedSeq to build SphericalHarmonicsLight")
```

Comment:
Name the parameter not the parameter type in the exception. IndexedSeq → coefficients